src/SFA.DAS.EmployerFinance.UnitTests/Models/AccountTests.cs

```diff
@@ -0,0 +1,223 @@
+            AccountPayeScheme.SetupGet(aps => aps.AccountId).Returns(Account.Id);
+            AccountPayeScheme.SetupGet(aps => aps.EmployerReferenceNumber).Returns(EmployerReferenceNumber);
+        }
+        
+        public AccountTestsFixture UpdatedPreviously()
+        {
+            Account.Updated = _fixture.Create<DateTime>();
+            ActionDate = Account.Updated.Value.AddMinutes(1);
+            return this;
+        }
+
+        public AccountTestsFixture UpdatedWithFutureUpdate()
+        {
+            Account.Updated = _fixture.Create<DateTime>();
+            ActionDate = Account.Updated.Value.AddMinutes(-1);
+            return this;
+        }
+
+        /// <summary>
+        /// Clones public properties. Leaves _accountPayeSchemes alone: it's not public and will contain mock AccountPayeScheme, for which cloning makes no sense.
+        /// </summary>
+        public AccountTestsFixture PartCloneOriginalAccount()
+        {
+            OriginalAccount = new Account
+            {
+                Id = Account.Id,
+                Name = Account.Name,
+                Created = Account.Created,
+                Updated = Account.Updated,
+                HashedId = Account.HashedId,
+                PublicHashedId = Account.PublicHashedId
+            };
+            return this;
+        }
+
+        public AccountTestsFixture AddPreExistingPayeScheme()
+        {
+            Account._accountPayeSchemes.Add(AccountPayeScheme.Object);
+            return this;
+        }
+        
+        #endregion Arrange
+
+        #region Act
+        
+        public void UpdateName(string name = NewName)
+        {
+            PartCloneOriginalAccount();
+            Account.UpdateName(name, ActionDate);
+        }
+
+        public AccountPayeScheme AddPayeScheme(string employerReferenceNumber = EmployerReferenceNumber)
+        {
+            PartCloneOriginalAccount();
+            return Account.AddPayeScheme(employerReferenceNumber, ActionDate);
+        }
+
+        public void RemovePayeScheme()
+        {
+            PartCloneOriginalAccount();
+            Account.RemovePayeScheme(AccountPayeScheme.Object, ActionDate);
+        }
+        
+        #endregion Act
+
+        #region Assert
+
+        public AccountTestsFixture AssertNameAndUpdatedDateUpdated()
+        {
+            Account.Name.Should().Be(NewName);
+            Account.Updated.Should().Be(ActionDate);
+            return this;
+        }
+
+        public AccountTestsFixture AssertNameAndUpdatedDateNotUpdated()
+        {
+            Account.Name.Should().Be(OriginalAccount.Name);
+            Account.Updated.Should().Be(OriginalAccount.Updated);
+            return this;
+        }
+
+        public AccountTestsFixture AssertPayeSchemeAdded()
+        {
+            // assumes no transformation in AccountPayeScheme c'tor. can we remove that assumption, but still test that the correct AccountPayeScheme is added?
```

Comment:
Those properties in `AccountPayeScheme` are set in its constructor. Ideally a properly isolated test for `Account` shouldn't rely on what another class happens to do in one of its methods (in this case, `AccountPayeScheme`'s c'tor).

To illustrate this, image if `AccountPayeScheme`'s constructor was changed to set its `Created` property to the supplied created date + 1 hour (for some strange reason). That would break the `Account` test, even though the `Account` class is behaving correctly.. Bit of a stretch I know, which is why the test relies on the constructor not doing anything funky, but that's what the comment refers to 🙂.

Reply:
Ok, I see, I think that is fine because if we do decide to do something funky in the constructor, a failing test will prompt us to think twice.